docs/taxonomy-revamp-2025.md

```diff
@@ -0,0 +1,93 @@
+- YAML is a notoriously complex, loose format with a lot of potholes.
+  - YAML files of different specifications parse completely differently (e.g., 1.2 vs 1.1).
+    - Note that PyYAML, our base tool [^2], parses YAML 1.1, not 1.2. There is a long way to go[^3] to support 1.2, which has been the latest spec since 2009. As such, even if someone were to search the Internet for a solution because they are not familiar with YAML, they likely will stumble across 1.2 solutions that don't work for 1.1.
+  - There are at least 9 different ways to indicate a multi-line string in YAML[^4], depending on which block scalar indicator[^5] is used and which block chomping indicator[^6] is used (this does **not** count the indentation indicator[^7]!). Then there are double-quoted flow scalar multilines[^8] and single-quoted flow scalar multilines[^9], which can cause more problems.
+- The linting system, intended to ensure the YAML file is readable by the SDG process, adds more burden on the non-technical user.
+  - The linter for YAML enforces an 80-character line length by default. That makes sense if you're working on code read from a terminal, but not to a typical end user used to working with rich text editors for a reading comprehension experience working with paragraphs.
+  - The linter also complains about trailing whitespace, another common thing that the typical end user won't understand why everything is failing.
+
+From a code perspective,
+
+- We are already using JSON in the data mixing process in SDG[^10].
+- Docling also exports JSON as input and output[^11].
+- JSON is also much more friendly to UI work, which is a primary path we would like people to use.
+
+Overall, the `qna.yaml` file needs to have fewer knobs and fewer pitfalls.
+
+The process of writing question and answer sets also is more like writing reading comprehension sets from a standardized exam. It would be better to frame this hands-on part of the process as similar to the passage and question sets from English reading comprehension exams
+
+## Proposed solutions
+
+To fix the user experience when working with data, I propose the following ideas. In general, the basic idea is "Keep It Simple; Make It Tick."
+
+### Use a schema field rather than directory tree structure
+
+Drop the folder structure in favor of a schema field for submission type and even domain, if necessary. The schema field can be entered automatically via the UI through a user selecting `knowledge` or `skill`.
+
+### Streamline the schema
+
+Make `created_by`, `domain`, and `document_outline` optional fields. Enforce those fields for the upstream taxonomy though documentation, CI, and review rather than require them for everyone.
+
+### Switch to JSON and Markdown for the `qna.yaml` document
+
+Allow the user to use Markdown in a WYSIWYG experience, and then use a Markdown-to-JSON converter to handle the conversion to a code-friendly format.
```

Comment:
I am a little worried that this would lead to a situation where it is easy to write Markdown that looks good but is not usable by SDG after it goes through the Markdown-to-JSON converter.  However, I am not sure how big a deal that would be.  What might help provide some intuitions is an example of what such a Markdown would look like; that might provide more of a sense of how hard it is to write SDG-complient Markdown.

Reply:
(Also, FWIW, I agree with the premise here that YAML is a big part of the problem with our current taxonomy format.)